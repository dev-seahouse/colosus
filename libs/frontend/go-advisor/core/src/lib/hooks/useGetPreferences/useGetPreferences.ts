import type { ConnectAdvisorGetPreferencesResponseDto } from '@bambu/api-client';
import { ConnectAdvisorPreferencesApi } from '@bambu/api-client';
import type { QueryClient, UseQueryOptions } from '@tanstack/react-query';
import { useQuery } from '@tanstack/react-query';

// get leads filter settings
export function getPreferencesQuery() {
  return {
    queryKey: ['getPreferences'],
    queryFn: fetchPreferences,
  };
}
async function fetchPreferences() {
  const connectAdvisorPreferencesApi = new ConnectAdvisorPreferencesApi();
  const res = await connectAdvisorPreferencesApi.getPreferences();
  return res.data;
}
export function useGetPreferences<
  TQueryReturnData = ConnectAdvisorGetPreferencesResponseDto
>(
  options?: Omit<
    UseQueryOptions<
      ConnectAdvisorGetPreferencesResponseDto,
      unknown,
      TQueryReturnData
    >,
    'queryFn'
  >
) {
  return useQuery({
    ...getPreferencesQuery(),
    ...options,
  });
}

export function getPreferencesLoader(queryClient: QueryClient) {
  const query = getPreferencesQuery();
  return async () => {
    return (
      // return data or fetch it
      queryClient.getQueryData(query.queryKey) ??
      (await queryClient.fetchQuery(query))
    );
  };
}

export default useGetPreferences;
