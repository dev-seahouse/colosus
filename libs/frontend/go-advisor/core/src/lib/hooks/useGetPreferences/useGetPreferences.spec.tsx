import {
  renderHook,
  queryClientWrapper,
  waitFor,
} from '@bambu/react-test-utils';
import useGetPreferences from './useGetPreferences';

describe('useGetPreferences', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(() => useGetPreferences(), {
      wrapper: queryClientWrapper,
    });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
