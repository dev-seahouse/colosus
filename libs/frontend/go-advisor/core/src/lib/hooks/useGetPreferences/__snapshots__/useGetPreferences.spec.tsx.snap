// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGetPreferences > should render successfully 1`] = `
{
  "createdAt": "2023-12-17T03:24:00",
  "minimumAnnualIncomeThreshold": 111111,
  "minimumRetirementSavingsThreshold": 222222,
  "updatedAt": "2023-12-17T03:24:00",
}
`;
